extensions/podman/packages/extension/src/podman-configuration.ts

```diff
@@ -30,24 +30,26 @@ const configurationRosetta = 'setting.rosetta';
  * Manages access to the containers.conf configuration file used to configure Podman
  */
 export class PodmanConfiguration {
+  updatedProxySettingsPromise: Promise<void> | undefined = undefined;
+
   async init(): Promise<void> {
     let httpProxy = undefined;
     let httpsProxy = undefined;
     let noProxy = undefined;
 
     // we receive an update for the current proxy settings
     extensionApi.proxy.onDidUpdateProxy(async (proxySettings: ProxySettings) => {
-      await this.updateProxySettings(proxySettings);
+      await this.updateProxySettigsSync(proxySettings);
```

Comment:
The `*Sync` method name is usually reserved for blocking io, see node method `existsSync`  :/ so having a await is a bit weird